Source/Request.swift

```diff
@@ -650,8 +706,8 @@ extension Request: CustomDebugStringConvertible {
 public protocol RequestDelegate: AnyObject {
     var sessionConfiguration: URLSessionConfiguration { get }
 
-    func willAttemptToRetryRequest(_ request: Request) -> Bool
-    func retryRequest(_ request: Request, ifNecessaryWithError error: Error)
+    func retryResult(for request: Request, dueTo error: Error, completion: @escaping (RetryResult?) -> Void)
+    func retryRequest(_ request: Request, withDelay timeDelay: TimeInterval?)
```

Comment:
These APIs are so much more explicit than the previous iterations. Now you simply request a retry result, or retry the request. Both are very clear as to what they actually do. Previously, it was pretty hard to tell what did what.